src/uniformscalingmap.jl

```diff
@@ -19,20 +19,30 @@ LinearAlgebra.transpose(A::UniformScalingMap) = A
 LinearAlgebra.adjoint(A::UniformScalingMap)   = UniformScalingMap(conj(A.λ), size(A))
 
 # multiplication with vector
+Base.:(*)(A::UniformScalingMap, x::AbstractVector) =
+    length(x) == A.M ? A.λ * x : throw(DimensionMismatch("A_mul_B!"))
+
 # call of LinearAlgebra.generic_mul! since order of arguments in mul! in stdlib/LinearAlgebra/src/generic.jl
 # TODO: either leave it as is or use mul! (and lower bound on version) once fixed in LinearAlgebra
-A_mul_B!(y::AbstractVector, A::UniformScalingMap, x::AbstractVector) =
-    (length(x) == length(y) == A.M ? LinearAlgebra.generic_mul!(y, A.λ, x) : throw(DimensionMismatch("A_mul_B!")))
-Base.:(*)(A::UniformScalingMap, x::AbstractVector) = A.λ * x
-
-At_mul_B!(y::AbstractVector, A::UniformScalingMap, x::AbstractVector) =
-    (length(x) == length(y) == A.M ? LinearAlgebra.generic_mul!(y, A.λ, x) : throw(DimensionMismatch("At_mul_B!")))
-
-Ac_mul_B!(y::AbstractVector, A::UniformScalingMap, x::AbstractVector) =
-    (length(x) == length(y) == A.M ? LinearAlgebra.generic_mul!(y, conj(A.λ), x) : throw(DimensionMismatch("Ac_mul_B!")))
+function A_mul_B!(y::AbstractVector, A::UniformScalingMap, x::AbstractVector)
+    (length(x) == length(y) == A.M || throw(DimensionMismatch("A_mul_B!")))
+    if iszero(A.λ)
+        return fill!(y, 0)
+    elseif isone(A.λ)
+        return copyto!(y, x)
+    else
+        return LinearAlgebra.generic_mul!(y, A.λ, x)
+    end
+end
+At_mul_B!(y::AbstractVector, A::UniformScalingMap, x::AbstractVector) = A_mul_B!(y, transpose(A), x)
+Ac_mul_B!(y::AbstractVector, A::UniformScalingMap, x::AbstractVector) = A_mul_B!(y, adjoint(A), x)
 
 # combine LinearMap and UniformScaling objects in linear combinations
-Base.:(+)(A1::LinearMap, A2::UniformScaling{T}) where {T} = A1 + UniformScalingMap{T}(A2[1,1], size(A1, 1))
-Base.:(+)(A1::UniformScaling{T}, A2::LinearMap) where {T} = UniformScalingMap{T}(A1[1,1], size(A2, 1)) + A2
-Base.:(-)(A1::LinearMap, A2::UniformScaling{T}) where {T} = A1 - UniformScalingMap{T}(A2[1,1], size(A1, 1))
-Base.:(-)(A1::UniformScaling{T}, A2::LinearMap) where {T} = UniformScalingMap{T}(A1[1,1], size(A2, 1)) - A2
+Base.:(+)(A1::LinearMap{T}, A2::UniformScaling) where {T} =
+    A1 + UniformScalingMap(convert(promote_type(T, eltype(A2)), A2.λ), size(A1, 1))
```

Comment:
Looks good, but still a bit asymmetric though. It could just as well be:
```julia
Base.:(+)(A1::LinearMap, A2::UniformScaling =
	    A1 + UniformScalingMap(convert(promote_type(eltype(A1), eltype(A2)), A2.λ), size(A1, 1))
```